listenbrainz/db/feedback.py

```diff
@@ -0,0 +1,152 @@
+            recording_msid: the MessyBrainz ID of the recording
+    """
+
+    with db.engine.connect() as connection:
+        connection.execute(sqlalchemy.text("""
+            DELETE FROM recording_feedback
+             WHERE user_id = :user_id
+               AND recording_msid = :recording_msid
+            """), {
+                'user_id': user_id,
+                'recording_msid': recording_msid,
+            }
+        )
+
+
+def get_feedback(user_id: int):
+    """ Get a list of recording feedbacks given by the user.
+        Args:
+            user_id: the row ID of the user in the DB
+        Returns:
+            A list of user's loved and hated recordings
+    """
+    records = []
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT user_id, recording_msid, score
+              FROM recording_feedback
+          ORDER BY created DESC
+            """), {
+                'user_id': user_id
+            }
+        )
+        while True:
+            row = result.fetchone()
+            if not row:
+                break
+
+            records.append(_format_recording_feedback_row(row[0], row[1], row[2]))
+
+    return records
+
+
+def get_feedback_single_type(user_id: int, score: int):
+    """ Get a list of recording feedbacks of particular type (loved or hated) given by the user.
+        If score is +1 then get loved recordings, if -1 then get hated recordings.
+        Args:
+            user_id: the row ID of the user
+        Returns:
+            A list of user's loved or hated recordings
+    """
+
+    records = []
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT user_id, recording_msid, score
+              FROM recording_feedback
+             WHERE user_id = :user_id
+               AND score = :score
+          ORDER BY created DESC
+            """), {
+                'user_id': user_id,
+                'score': score
+            }
+        )
+        while True:
+            row = result.fetchone()
+            if not row:
+                break
+
+            records.append(_format_recording_feedback_row(row[0], row[1], row[2]))
+
+    return records
+
+
+def get_feedback_for_recording(recording_msid: str):
+    """ Get a list of feedbacks for a given recording.
+        Args:
+            recording_msid: the MessyBrainz ID of the recording
+        Returns:
+            A list of feedbacks for the given recording
+    """
+
+    records = []
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT user_id, recording_msid, score
+              FROM recording_feedback
+             WHERE recording_msid = :recording_msid
+          ORDER BY created DESC
+            """), {
+                'recording_msid': recording_msid
+            }
+        )
+        while True:
```

Comment:
same here, let's do it how it's done in other places.

Reply:
Done